File: app/src/main/java/com/verbatoria/di/dashboard/DashboardComponent.java
package com.verbatoria.di.dashboard;

import com.verbatoria.ui.calendar.view.CalendarFragment;
import com.verbatoria.ui.dashboard.view.DashboardActivity;
import com.verbatoria.ui.dashboard.view.info.VerbatologInfoFragment;
import com.verbatoria.ui.dashboard.view.settings.SettingsFragment;

import dagger.Subcomponent;

/**
 * Компонент Даггера для dashboard
 *
 * @author nikitaremnev
 */
@Subcomponent(modules = {DashboardModule.class})
@DashboardScope
public interface DashboardComponent {

    void inject(DashboardActivity dashboardActivity);

    void inject(VerbatologInfoFragment verbatologInfoFragment);

    void inject(CalendarFragment calendarFragment);

    void inject(SettingsFragment settingsFragment);

}

File: app/src/main/java/com/verbatoria/di/dashboard/DashboardComponent.kt
package com.verbatoria.di.dashboard

import com.verbatoria.di.BaseInjector
import com.verbatoria.di.dashboard.calendar.CalendarComponent
import com.verbatoria.di.dashboard.info.InfoComponent
import com.verbatoria.di.dashboard.settings.SettingsComponent
import com.verbatoria.ui.dashboard.DashboardActivity
import dagger.Subcomponent

/**
 * @author n.remnev
 */

@Subcomponent(modules = [DashboardModule::class])
interface DashboardComponent : BaseInjector<DashboardActivity> {

    fun plusInfoComponent(): InfoComponent.Builder

    fun plusCalendarComponent(): CalendarComponent.Builder

    fun plusSettingsComponent(): SettingsComponent.Builder

    @Subcomponent.Builder
    interface Builder {

        fun build(): DashboardComponent

    }

}
